rw_tlv.m4

```diff
@@ -355,131 +355,71 @@ m4_define(['m4_into_fields'], ['{m4_into_fields_lhs(['$2'], [''], $1_FIELDS)['}
-m4_popdef(['m4_MAX'])
-m4_popdef(['m4_MIN'])
-m4_popdef(['m4_sig'])
-m4_popdef(['m4_init'])
-m4_popdef(['m4_op'])
-m4_popdef(['m4_cond_expr'])
-'])
+      $_redux_sig = $_init;
+      for (int i = #_MIN; i <= #_MAX; i++)
+         $_redux_sig = $_redux_sig #_op /_hier[i]$_sig;
 
 
 // Select across a hierarchy (MUX) within a pipeline with a decoded select.  Works for $pipe_signals and $ANY.
-// m4+select($selected_value[7:0], >top, >hier, ...fix)
-m4_define_plus(['m4_select'], ['
-
-m4_pushdef(['m4_redux_sig'],    ['$5'])   // The resulting signal, including bit range.
-m4_pushdef(['m4_top'],          ['$6'])   // Base scope for references.
-m4_pushdef(['m4_hier'],         ['$7'])   // use m4_top['']m4_hier[*]m4_subhier
-m4_pushdef(['m4_subhier'],      ['$8'])   // Replicated logic is created under m4_hier[*].
+// m4+select($selected_value[7:0], /top, /hier, ...fix)
+//$_redux_sig The resulting signal, including bit range.
+//  /_top Base scope for references.
+// /_hier  use /_top['']/_hier[*]/_subhier
+// /_subhier Replicated logic is created under m4_hier[*].
 //m4_pushdef(['m4_MAX'],          ['$8'])
 //m4_pushdef(['m4_MIN'],          ['$9'])
-m4_pushdef(['m4_sel_sig_subhier'],['$9'])   // m4_top['']m4_hier[*]m4_sel_sig_subhier['']m4_sel_sig selects an input.
-m4_pushdef(['m4_sel_sig'],      ['$10'])
-m4_pushdef(['m4_sig'],          ['$11'])  // The signal to select, including bit range.
-m4_pushdef(['m4_redux_sig_cond'], ['$12'])  // When condition for redux sig.  [''] for none (produces '0 if no select);  ['-'] or ['-$signame'] to generate as "| m4_top['']m4_hier[*]m4_sel_sig_subhier['']m4_sel_sig"; ['$signame'] to use given signal.
+// /_sel_sig_subhier /_top['']/_hier[*]$_sel_sig_subhier['']$_sel_sig selects an input.
+// $_redux_sig_cond    When condition for redux sig.  [''] for none (produces '0 if no select);  ['-'] or ['-$signame'] to generate as "| /_top['']/_hier[*]/_sel_sig_subhier['']/_sel_sig"; ['$signame'] to use given signal.
+\TLV select($_redux_sig,/_top,/_hier,/_subhier,$_sel_sig_subhier,$_sel_sig,$_sig,$_redux_sig_cond)
 
-m4_pushdef(['m4_hier_index'], ['m4_substr(m4_hier, 1)'])
+m4_pushdef(['m4_hier_index'], ['m4_substr(/_hier, 1)'])
 m4_pushdef(['m4_assign_redux_sig_cond'], m4_ifelse(m4_substr(m4_redux_sig_cond, 0, 1), ['-'], ['true'], ['']))  // Has '-'.
-m4_define(['m4_redux_sig_cond'],         m4_ifelse(m4_substr(m4_redux_sig_cond, 0, 1), ['-'], m4_substr(m4_redux_sig_cond, 1), m4_redux_sig_cond))  // Remove '-'.
-m4_define(['m4_redux_sig_cond'],         m4_ifelse(m4_redux_sig_cond,[''],m4_sel_sig['']_cond,m4_redux_sig_cond))  // Make up a signal name if not provided.
-
-'], m4___file__, m4___line__, ['
+m4_define(['$_redux_sig_cond'],         m4_ifelse(m4_substr($_redux_sig_cond, 0, 1), ['-'], m4_substr($_redux_sig_cond, 1), $_redux_sig_cond))  // Remove '-'.
+m4_define(['$_redux_sig_cond'],         m4_ifelse($_redux_sig_cond,[''],$_sel_sig['']_cond,$_redux_sig_cond))  // Make up a signal name if not provided.
```

Comment:
m4_define(['S_redux_sig_cond'],         m4_ifelse(S_redux_sig_cond,[''],$_sel_sig['']_cond,S_redux_sig_cond))  // Make up a signal name if not provided.

...and use S_redux_sig_cond below